src/main/java/org/openrewrite/hibernate/MigrateBooleanMappings.java

```diff
@@ -0,0 +1,112 @@
+        REPLACEMENTS.put("boolean", "NumericBooleanConverter");
+    }
+
+    @Override
+    public String getDisplayName() {
+        return "Replace boolean type mappings with converters";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Replaces type mapping of booleans with appropriate attribute converters.";
+    }
+
+    @Override
+    public TreeVisitor<?, ExecutionContext> getVisitor() {
+        return Preconditions.check(new UsesType<>("org.hibernate.annotations.Type", true),
+                new JavaIsoVisitor<ExecutionContext>() {
+                    @Override
+                    public J.Annotation visitAnnotation(J.Annotation annotation, ExecutionContext ctx) {
+                        J.Annotation ann = super.visitAnnotation(annotation, ctx);
+                        if (!TypeUtils.isOfClassType(ann.getType(), "org.hibernate.annotations.Type")) {
+                            return ann;
+                        }
+
+                        List<Expression> args = ann.getArguments();
+                        if (args == null) {
+                            return ann;
+                        }
+
+                        Object type = args.stream()
+                                .filter(exp -> {
+                                    if (exp instanceof J.Assignment) {
+                                        J.Identifier variable = (J.Identifier) ((J.Assignment) exp).getVariable();
+                                        return "type".equals(variable.getSimpleName());
+                                    }
+                                    return false;
+                                })
+                                .findFirst()
+                                .map(exp -> {
+                                    Expression value = ((J.Assignment) exp).getAssignment();
+                                    if (value instanceof J.Literal) {
+                                        return ((J.Literal) value).getValue();
+                                    }
+                                    return null;
+                                })
+                                .orElse(null);
+
+                        if (type instanceof String && REPLACEMENTS.containsKey((String) type)) {
+                            String converterName = REPLACEMENTS.get((String) type);
+                            String converterFQN = String.format("org.hibernate.type.%s", converterName);
+
+                            ann = JavaTemplate.builder(String.format("@Convert(converter = %s.class)", converterName))
+                                    .javaParser(JavaParser.fromJavaVersion().classpath("hibernate-core", "jakarta.persistence-api"))
+                                    .imports(converterFQN, "jakarta.persistence.Convert")
+                                    .contextSensitive()
+                                    .build().apply(getCursor(), ann.getCoordinates().replace());
```

Comment:
I'm guessing it's rare to non-existent for there to be additional parameters for these types? Just to be sure nothing is lost when we replace the annotation here.

Reply:
The boolean types have no additional parameters, so they'd have no effect.

However, it's a good thing you asked, as it made me look into the implementation. The mapping for `"boolean"` is dialect-specific, so replacing it with the NumericBooleanConverter may not always be correct. The old equivalent for that converter is `"numeric_boolean"`.

What should we do with `"boolean"`? Assume numeric conversion works (it probably does for most dialect), or not replace the annotation at all?

Reply:
The code would not compile, as `@Type` expects a class now instead of a string. So I'll add support for `"numeric_boolean"` and remove the replacement for `"boolean"`.

Reply:
Great, thanks a lot for those changes; we should be good to merge then and can always follow up with more specific handling for boolean if needed in the future.